pixl_imaging/src/pixl_imaging/_processing.py

```diff
@@ -143,27 +144,112 @@ async def _add_project_to_study(
         )
 
 
-async def _find_study_in_vna_or_raise(orthanc_raw: Orthanc, study: ImagingStudy) -> str:
+async def _find_study_in_archives_or_raise(orthanc_raw: Orthanc, study: ImagingStudy) -> str:
     """
-    Query the VNA for the study using its UID, fallback to querying on MRN + accession number if
-    UID is not available, raise exception if it doesn't exist
+    Query primary and secondary archives for a study.
+
+    - Query the primary archive for the study using its UID. If UID is not available, query on
+      MRN and accession number.
+    - If not found in the primary archive, and it is a weekday between 8 pm and 8 am, query the
+      online secondary archive using the study UID. If UID is not available, query on
+      MRN + accession number.
+    - If not found in the online secondary archive, query the nearline secondary archive. If found
+      in the nearline, assume the study is in the primary archive and log it.
+    - If not found in either the primary archive or the online secondary archive,
+      raise a PixlDiscardError.
     """
-    query_id = await orthanc_raw.query_remote(
-        study.orthanc_uid_query_dict, modality=config("VNAQR_MODALITY")
+    query_id = await _find_study_in_archive(
+        orthanc_raw=orthanc_raw,
+        study=study,
+        modality=config("PRIMARY_DICOM_MODALITY"),
     )
-    if query_id is None:
-        logger.info(
-            "No study found with UID {}, trying MRN and accession number", study.message.study_uid
-        )
-        query_id = await orthanc_raw.query_remote(
-            study.orthanc_query_dict, modality=config("VNAQR_MODALITY")
+
+    if query_id is not None:
+        return str(query_id)
+    if _is_daytime() or _is_weekend():
+        msg = (
+            f"Failed to find study {study.message.study_uid} in primary archive. "
+            "Not querying secondary archive during the daytime or on the weekend."
         )
+        raise PixlDiscardError(msg)
+
```

Comment:
should also return early if the secondary dicom server isn't defined (say AE title?), so that we don't have to use the PACS

Reply:
ah I see, I've done this now, rasing an error if `SECONDARY_DICOM_AE_TITLE` is undefined / an empty string

Reply:
yeah reasonable way around this